pkg/scalers/selenium_grid_scaler.go

```diff
@@ -0,0 +1,201 @@
+package scalers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+	v2beta2 "k8s.io/api/autoscaling/v2beta2"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+)
+
+type seleniumGridScaler struct {
+	metadata *seleniumGridScalerMetadata
+	client   *http.Client
+}
+
+type seleniumGridScalerMetadata struct {
+	url         string
+	browserName string
+	targetValue int64
+}
+
+type seleniumResponse struct {
+	Data data `json:"data"`
+}
+
+type data struct {
+	SessionsInfo sessionsInfo `json:"sessionsInfo"`
+}
+
+type sessionsInfo struct {
+	SessionQueueRequests []string          `json:"sessionQueueRequests"`
+	Sessions             []seleniumSession `json:"sessions"`
+}
+
+type seleniumSession struct {
+	Id           string `json:"id"`
+	Capabilities string `json:"capabilities"`
+	NodeId       string `json:"nodeId"`
+}
+
+type capability struct {
+	BrowserName string `json:"browserName"`
+}
+
+func NewSeleniumGridScaler(config *ScalerConfig) (Scaler, error) {
+	meta, err := parseSeleniumGridScalerMetadata(config)
+
+	if err != nil {
+		return nil, fmt.Errorf("error parsing selenium grid metadata: %s", err)
+	}
+
+	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout)
+
+	return &seleniumGridScaler{
+		metadata: meta,
+		client:   httpClient,
+	}, nil
+}
+
+func parseSeleniumGridScalerMetadata(config *ScalerConfig) (*seleniumGridScalerMetadata, error) {
+	meta := seleniumGridScalerMetadata{
+		targetValue: 1,
+	}
+
+	if val, ok := config.TriggerMetadata["url"]; ok {
+		meta.url = val
+	} else {
+		return nil, fmt.Errorf("no selenium grid url given in metadata")
+	}
+
+	if val, ok := config.TriggerMetadata["browserName"]; ok {
+		meta.browserName = val
+	} else {
+		return nil, fmt.Errorf("no browser name given in metadata")
+	}
+
+	return &meta, nil
+}
+
+// No cleanup required for selenium grid scaler
+func (s *seleniumGridScaler) Close() error {
+	return nil
+}
+
+func (s *seleniumGridScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	v, err := s.getSessionsCount()
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, fmt.Errorf("error requesting selenium grid endpoint: %s", err)
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *v,
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *seleniumGridScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetValue := resource.NewQuantity(s.metadata.targetValue, resource.DecimalSI)
+	metricName := kedautil.NormalizeString(fmt.Sprintf("%s-%s", "seleniumgrid", s.metadata.browserName))
```

Comment:
`metricName` should be unique, so in case you define mutliple `triggers` in one `ScaledObject` the `metricName` for each trigger is unique. In this case if I've defined mutliple `firefox` triggers pointing to different selenium grids, the triggers would have same `metricName`.

Reply:
I just added the browserVersion to be appended to the metricName